src/components/MissionsProfile.js

```diff
@@ -0,0 +1,21 @@
+import { useSelector } from 'react-redux';
+
+const MissionsProfile = () => {
+  const missions = useSelector((state) => state.missions.missions);
+  const reservedMissions = missions.filter((x) => x.reserved);
+
+  return (
+    <div className="reserve-container">
+      <h1 className="profile-missions-title">My Missions</h1>
+      <div className="profile-missions">
+        {reservedMissions.length === 0 && <p className="no-reserve">No Mission Reserved</p>}
+        {reservedMissions.map((x) => (
+          <div key={x.mission_id} className="reserve-item">{x.mission_name}</div>
+        ))}
+
+      </div>
```

Comment:
- [ ] Impressive job Team. Here you have used the `map method` to map the `reservedMissions` this gets the job done and works fine. But if you think about it `map` should be used when we need to map some sort of data to another, (let's say you would like to add a salary of 5% to all the employees in your array `here map makes sense`). But the `filter method` should be used to find some sort of data that are different in some way, (Let's say you would like to find an employee whose salary is >=5000$ `here filter makes sense`). This is to say that you should use a filter instead of a map to find `reservedMissions`. 
> This is one of the main requirements we look into the projects.

Reply:
The code is already using filter to filter the missions that have been reserved:
````
const reservedMissions = missions.filter((x) => x.reserved);
````

To use filter instead of map for rendering the reserved missions, we have modified the code as follows:

````
<div className="profile-missions">
  {reservedMissions.length === 0 && <p className="no-reserve">No Mission Reserved</p>}
  {reservedMissions.filter((x) => x.mission_id !== undefined).map((x) => (
    <div key={x.mission_id} className="reserve-item">{x.mission_name}</div>
  ))}
</div>
````
In this modified code, we are using filter to remove any elements in the reservedMissions array that have an undefined mission_id. The filtered array is then passed to map to generate the JSX for each reserved mission.
However, it's worth noting that in most cases, reserved missions should always have a defined mission_id. If there are reserved missions with undefined mission_id, it may be worth investigating and fixing the underlying issue that's causing this.

I iwill give it more time during my free time when i am not on pressure of finishing the capstone,Otherwise thank you @TadesseDev for your wonderful reviews